File: leetcode-new/62.cpp
class Solution
{
public:
    int uniquePaths(int m, int n)
    {
        vector<int> prev(n, 1);
        vector<int> curr(n, 0);

        for (int r = 1; r < m; ++r)
        {
            curr[0] = 1;
            for (int c = 1; c < n; ++c)
            {
                curr[c] = prev[c] + curr[c - 1];
            }
            prev = move(curr);
            curr = vector<int>(n, 0);
        }
        return prev[n - 1];
    }
};

File: leetcode/62.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [[0 for j in range(n)] for i in range(m)]
        dp[0][0] = 1
        rows, cols = m, n

        for r in range(rows):
            for c in range(cols):
                if 0 <= r - 1 < rows:
                    dp[r][c] += dp[r - 1][c]
                if 0 <= c - 1 < cols:
                    dp[r][c] += dp[r][c - 1] 
                
        return dp[-1][-1]
